### ISSUE TITLE
Performance of yolov8 vs yolov5 when exported as ONNX

### ISSUE THREAD

**Issue by JustasBart** [ultralytics/ultralytics#759]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi guys,

With the new possibility of #721 I was doing some ONNX testing and one thing came to my attention immediately was the performance of yolov8 models vs yolov5 models...

I'm comparing yolov8s.onnx and yolov5s.onnx exported with the same configurations:

yolov5:
```
python3 export.py \
--weights yolov5s.pt \
--img 480 640 \
--include onnx \
--opset 12
```
yolov8:
```
yolo export \
model=yolov8s.pt \
imgsz=[480,640] \
format=onnx \
opset=12
```

Both models are ran under the same configuration of (0.25 conf-thres, 0.45 iou-thres, 0.5 nms-thres)(w:640, h:480):

yolov5s.onnx performance:
![image](https://user-images.githubusercontent.com/40023722/216091027-673ed848-ff09-46d9-a269-4d5af613b0ba.png)
![image](https://user-images.githubusercontent.com/40023722/216091073-bbb28220-71d1-40ba-97c1-128f2217d8f1.png)

As you can see it did pretty well, it only missed one tie.

yolov8s.onnx performance:
![image](https://user-images.githubusercontent.com/40023722/216090641-fc526cc1-484c-456a-8519-7329d56f4ef3.png)
![image](https://user-images.githubusercontent.com/40023722/216090700-9c344006-df6d-496e-ac54-8cdc28915fac.png)

As you can see v8 missed both ties and even the bus.

However, when I run the experiment on the original PyTorch (.pt) models (Image-size: 640x480, else defaults) I can see the following results:

yolov5.pt performance:
![zidane](https://user-images.githubusercontent.com/40023722/216088758-7b8a481d-5678-4cd4-b52e-9cd30248e574.jpg)
![bus](https://user-images.githubusercontent.com/40023722/216088780-f70410e4-1957-4f26-81b9-9f613c224a06.jpg)

Note that it missed a person on the left for the bus image.

yolov8.pt performance:
![zidane](https://user-images.githubusercontent.com/40023722/216087944-cb0e3ed1-4e4d-42b2-bee7-1bc82de32e0c.jpg)
![bus](https://user-images.githubusercontent.com/40023722/216087775-8a503177-34c6-48b0-b615-33e8cf9b33a7.jpg)

Note that yolov8s.pt got two ties where there was only one and then also missed a tie (Maybe not used to 640x480, however, yolov5 did it just fine).

So my question is why do the .pt models perform generally better than the exported .onnx ones?
And most important why would the yolov8 exported .onnx models perform worse than the yolov5 exported .onnx models? Would you guys think it's more likely that it's to do with the export itself or the C++ code inferencing the .onnx models?

Thanks :rocket:

**Comment by triple-Mu** [comment]
The pytorch model use letterbox function auto＝true arguement when infer. It may resize the image with mini pad.

**Comment by JustasBart** [comment]
@triple-Mu Thanks for the reply, however I don't think that that's it, I've tested it with padding for yolov8s.onnx to see the following results:

![image](https://user-images.githubusercontent.com/40023722/216098970-8ffd2bf7-8094-4046-ba9a-2ba5ee5556a8.png)
![image](https://user-images.githubusercontent.com/40023722/216099001-ac015230-8d35-49ab-90c4-74cc3ba4a8b2.png)

![image](https://user-images.githubusercontent.com/40023722/216099041-57968236-6234-43e0-bfda-702f18509303.png)
![image](https://user-images.githubusercontent.com/40023722/216099072-8fb538fc-598b-402f-b98f-1cc0d8f6928a.png)

Basically the same setup but I've exported it as simply:
```
yolo export \
model=yolov8s.pt \
imgsz=640 \
format=onnx \
opset=12
```
And put it through the letterBox code just before inference...

**Comment by JustasBart** [comment]
Turns out my current yolov8 C++ will only run inference on class index 0 only (Oops) closing for now.

**Comment by ajeshmonr7777** [comment]
@JustasBart  This was the issue of the code. You might have used same Yolo v5 logic in code. Here we have 84 outputs not 85. That's why you were unable to predict other objects other than persons.

**Comment by JustasBart** [comment]
Thanks @ajeshmonr7777 I had finally finally figured it out moments ago!

![image](https://user-images.githubusercontent.com/40023722/217223423-5edac112-f2d5-45b9-88df-e2c2a2340fd2.png)
![image](https://user-images.githubusercontent.com/40023722/217223465-29d2b9d6-5c3a-4b5c-afa6-958bfefeea7a.png)

The repo is updated so anyone can use it now :rocket:
